app/src/main/java/org/schabi/newpipe/util/StreamDialogEntry.java

```diff
@@ -128,11 +134,13 @@
     private final int resource;
     private final StreamDialogEntryAction defaultAction;
     private StreamDialogEntryAction customAction;
+    private boolean requireNetwork;
 
     StreamDialogEntry(final int resource, final StreamDialogEntryAction defaultAction) {
         this.resource = resource;
         this.defaultAction = defaultAction;
         this.customAction = null;
+        this.requireNetwork = false;
     }
```

Comment:
Instead of manually using `.setRequireNetwork(true);` in every place the actions are used, I'd just change the constructor like this, since the actions that require network access are always the same.
```suggestion
    StreamDialogEntry(final int resource, final boolean requireNetwork, final StreamDialogEntryAction defaultAction) {
        this.resource = resource;
        this.requireNetwork = requireNetwork;
        this.defaultAction = defaultAction;
        this.customAction = null;
    }
```
If you do this, obviously remove the `setRequireNetwork` method.